server/src/main/java/org/opensearch/plugins/PluginsService.java

```diff
@@ -522,9 +527,16 @@ private static void addSortedBundle(
 
         dependencyStack.add(name);
         for (String dependency : bundle.plugin.getExtendedPlugins()) {
-            Bundle depBundle = bundles.get(dependency);
+            String dependencyName = dependency.split(";")[0];
```

Comment:
We cannot do that for bwc reasons:
 - please pre-parse in the `PluginInfo` (the `extendedPlugins` should be holding just plugin names as before)
 - please store optionality in separate list or it could be separate data structure, per your choice
 - please update `writeTo(final StreamOutput out)` / `PluginInfo(final StreamInput in)` to not send this additional optionality to the older nodes - they won't be able to interpret it properly

Just to reiterate the bwc part - we don't want "older" nodes to receive `dne;optional=true` (they would think this is a plugin name), we need to send `dne` only. Hope it makes sense.

Reply:
When is `PluginInfo` serialized and deserialized? I pushed a small change to exclude `;optional=true` when serializing PluginInfo and opted not to include any information about optionality when serializing/deserializing.

Reply:
Checking more of the calls to getExtendedPlugins()...

Reply:
Yep, `ClusterStatsNodes`